client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_bootstrap.c

```diff
@@ -0,0 +1,821 @@
+    }
+    return KAA_ERR_BADPARAM;
+}
+
+kaa_error_t ext_tcp_utils_tcp_socket_close(kaa_fd_t fd)
+{
+    if (fd != access_point_test_info.fd) {
+        return KAA_ERR_BADPARAM;
+    }
+    access_point_test_info.socket_disconnected_closed = true;
+    return KAA_ERR_NONE;
+}
+
+ext_tcp_socket_io_errors_t ext_tcp_utils_tcp_socket_read(kaa_fd_t fd, char *buffer, size_t buffer_size, size_t *bytes_read)
+{
+    if (fd != access_point_test_info.fd) {
+        return KAA_TCP_SOCK_IO_ERROR;
+    }
+    if (!buffer) {
+        return KAA_TCP_SOCK_IO_ERROR;
+    }
+    if (access_point_test_info.socket_connecting_error_scenario) {
+        *bytes_read = 0;
+        return KAA_TCP_SOCK_IO_ERROR;
+    } else if (!access_point_test_info.connack_read) {
+        memcpy(buffer,CONNACK,sizeof(CONNACK));
+        *bytes_read = sizeof(CONNACK);
+        access_point_test_info.connack_read = true;
+    } else if (!access_point_test_info.kaasync_read) {
+        memcpy(buffer,KAASYNC_BOOTSTRAP,sizeof(KAASYNC_BOOTSTRAP));
+        *bytes_read = sizeof(KAASYNC_BOOTSTRAP);
+        access_point_test_info.kaasync_read = true;
+    }
+
+    return KAA_TCP_SOCK_IO_OK;
+}
+
+ext_tcp_socket_io_errors_t ext_tcp_utils_tcp_socket_write(kaa_fd_t fd, const char *buffer, size_t buffer_size, size_t *bytes_written)
+{
+    if (fd != access_point_test_info.fd) {
+        return KAA_TCP_SOCK_IO_ERROR;
+    }
+    if (!buffer) {
+        return KAA_TCP_SOCK_IO_ERROR;
+    }
+    if (!access_point_test_info.auth_packet_written) {
+        if (buffer_size != (sizeof(CONNECT_HEAD)+sizeof(CONNECT_PACK))) {
+            return KAA_TCP_SOCK_IO_ERROR;
+        }
+        *bytes_written = 0;
+        if (!memcmp(buffer,CONNECT_HEAD,sizeof(CONNECT_HEAD))) {
+            if (!memcmp(buffer+sizeof(CONNECT_HEAD),CONNECT_PACK,sizeof(CONNECT_PACK))) {
+                access_point_test_info.auth_packet_written = true;
+                *bytes_written = sizeof(CONNECT_HEAD)+sizeof(CONNECT_PACK);
+                return KAA_TCP_SOCK_IO_OK;
+            }
+        }
+    } else if (!access_point_test_info.socket_disconnected_write) {
+        if (buffer_size != sizeof(DISCONNECT_MESSAGE)) {
+            return KAA_TCP_SOCK_IO_ERROR;
+        }
+        if (!memcmp(buffer,DISCONNECT_MESSAGE,sizeof(DISCONNECT_MESSAGE))) {
+            access_point_test_info.socket_disconnected_write = true;
+            *bytes_written = sizeof(DISCONNECT_MESSAGE);
+            return KAA_TCP_SOCK_IO_OK;
+        }
+    }
+
+    return KAA_TCP_SOCK_IO_ERROR;
+}
+
+kaatcp_error_t kaatcp_get_request_connect(const kaatcp_connect_t *message
+                                        , char *buf
+                                        , size_t *buf_size)
+{
+    if (message->protocol_name_length != KAA_TCP_NAME_LENGTH) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->protocol_version != PROTOCOL_VERSION) {
+        return KAATCP_ERR_BAD_PARAM;
+    }
+    if (message->next_ptorocol_id != 0x3553c66f) {
```

Comment:
The KAA_PLATFORM_PROTOCOL_ID constant from "kaa_platform_common.h" may be used instead.